### PR TITLE
#1846 eLL: eLL disappears when no slide is selected

### LINKED ISSUE #1846: eLL disappears when no slide is selected
Current: eLL panel disappears when no slide is selected. Furthermore, it is not possible to open eLL when no slide is selected.
image

Suggested: eLL panel should remain (but turn to blank?) even if a slide is not selected.

### PR DESCRIPTION
**Pull request by LiXiaoooowei**
Fixes #1846 

This PR changes eLL behavior when no slide is selected. Previous when slide selection is null, eLL automatically disappears and eLL button on the ribbon is disabled. This PR fixes eLL button on the ribbon to be enabled at all times. When slide selection is null, eLL panel still loads but its functions are disabled.

### CODE REVIEW

File: PowerPointLabs/PowerPointLabs/ThisAddIn.cs [hunk 1]
```diff
@@ -726,18 +726,18 @@ private void UpdateELearningPane(int selectedSlidesCount)
                 return;
             }
             ELearningLabTaskpane taskpane = elearningLabPane.Control as ELearningLabTaskpane;
-            if (selectedSlidesCount > 0)
-            {
+         //   if (selectedSlidesCount > 0)
```
Comment (ChesterSng) [inline PowerPointLabs/PowerPointLabs/ThisAddIn.cs]: Should these commented lines be removed?

Comment (LiXiaoooowei) [reply PowerPointLabs/PowerPointLabs/ThisAddIn.cs]: Thanks! Removed commented lines.

File: PowerPointLabs/PowerPointLabs/ELearningLab/Views/ELearningWorkspaceViews/ELearningLabMainPanel.xaml.cs [hunk 2]
```diff
@@ -91,26 +92,27 @@ public void ListViewItemPropertyChanged(object sender, PropertyChangedEventArgs
         /// </summary>
         public void ReloadELearningLabOnSlideSelectionChanged()
         {
-            PowerPointSlide _slide = this.GetCurrentSlide();
+            slide = this.GetCurrentSlide();
+            if (slide == null)
+            {
+                slideId = -1;
+                SetupSpinnerOnLoadingBegins("No slide is selected");
```
Comment (leeyh20) [inline PowerPointLabs/PowerPointLabs/ELearningLab/Views/ELearningWorkspaceViews/ELearningLabMainPanel.xaml.cs]: Could all these constant strings be placed in a ELearningText script file with constants?

File: PowerPointLabs/PowerPointLabs/ELearningLab/Views/ELearningWorkspaceViews/ELearningLabMainPanel.xaml.cs [hunk 3]
```diff
@@ -799,12 +806,15 @@ private bool DoesSlideExist(int slideId)
             return false;
         }
 
-        private void SetupSpinnerOnLoadingBegins()
+        private void SetupSpinnerOnLoadingBegins(string content)
```
Comment (leeyh20) [inline PowerPointLabs/PowerPointLabs/ELearningLab/Views/ELearningWorkspaceViews/ELearningLabMainPanel.xaml.cs]: Maybe the name `Begin` would be more consistent

File: PowerPointLabs/PowerPointLabs/ELearningLab/Views/ELearningWorkspaceViews/ELearningLabMainPanel.xaml.cs (hunk 2 above)
Comment (LiXiaoooowei) [reply PowerPointLabs/PowerPointLabs/ELearningLab/Views/ELearningWorkspaceViews/ELearningLabMainPanel.xaml.cs]: Thanks for the suggestion! Removed magic strings.

File: PowerPointLabs/PowerPointLabs/ELearningLab/Views/ELearningWorkspaceViews/ELearningLabMainPanel.xaml.cs (hunk 3 above)
Comment (LiXiaoooowei) [reply PowerPointLabs/PowerPointLabs/ELearningLab/Views/ELearningWorkspaceViews/ELearningLabMainPanel.xaml.cs]: Thanks! Renamed method name to `...Begin`

**Comment by YuPeiHenry** [comment]
@LiXiaoooowei It seems the merge conflict with ThisAddIn is not resolved correctly. The project does not build currently.

**Comment by LiXiaoooowei** [comment]
> Do you mind redoing the conflict resolution for ThisAddIn? The entire file was probably marked.

My bad. Re-resolved conflicts.

**Comment by leeyh20** [comment]
@ChesterSng Good job haha was just going to comment on the new line.

**Comment by LiXiaoooowei** [comment]
@ChesterSng Thanks haha!

**Pull request merged**